### ISSUE TITLE
Incorrect match for "Module names should be written in PascalCase."

### ISSUE THREAD

**Issue by blatyo** [rrrene/credo#340]
### Environment

* Credo version (`mix credo -v`): 0.7.0
* Erlang/Elixir version (`elixir -v`):
      
      Erlang/OTP 19 [erts-8.2] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

      Elixir 1.4.1
* Operating system: OS X 10.11.5

### What were you trying to do?

Run credo on a project. The project dynamically generates some modules in a macro. Similar code
to generate the warning, is the following:

``` elixir
x = X
defmodule x do
end
```

### Expected outcome

No match for "Module names should be written in PascalCase."

### Actual outcome

Match for "Module names should be written in PascalCase."

### Other notes

It's unclear why credo is checking for a variable in the name. It's not actually valid elixir to define a module with a lowercase bare word for it's name.

``` elixir
defmodule x do
end
# warning: variable "x" does not exist and is being expanded to "x()", please use parentheses to remove the ambiguity or change the variable name

# ** (CompileError) iex:4: undefined function x/0
#    (elixir) expanding macro: Kernel.defmodule/2
```

Perhaps this warning was trying to check for using a lower case atom for the name? For example:

``` elixir
defmodule :x do
end
```

Problematic code is here: https://github.com/rrrene/credo/blob/631a5fa4e98071892d0646ab376c7cf77848bbd9/lib/credo/check/readability/module_names.ex#L46-L47

I can fix this issue if it's agreed that it is an issue.

**Comment by rrrene** [comment]
@blatyo  Great bug report, correct analysis. Thx for the PR :tada: :+1:

**Comment by blatyo** [comment]
Should there also be a check for using lowercase atoms for modules? Happy to implement if so.

**Comment by rrrene** [comment]
@blatyo Thanks, but I don't think that's necessary.

To be honest, I am rethinking the necessity of this module-name-related check at the moment, since you were right in your initial comment: One can not create a module with a lower case name. This check might simply be a remnant from the very early days when Credo was just a proof-of-concept.

We just deprecated a couple of checks because they turned out not to be that useful and the `ModuleName` check seems to be another candidate.
